### PR TITLE
plan: fix a bug when do column substitution for join.

### PR DESCRIPTION
**Pull request by winoros**
Join's equal condition may become single side condition after substitution.
See added unit for more details.
PTAL @lamxTyler @zz-jason

### CODE REVIEW

File: executor/join_test.go [hunk 1]
```diff
@@ -626,6 +626,13 @@ func (s *testSuite) TestSubquery(c *C) {
 	tk.MustExec("INSERT INTO t1 (a) values(1), (2), (3), (4), (5)")
 	result = tk.MustQuery("select (select /*+ TIDB_INLJ(x1) */ x2.a from t1 x1, t1 x2 where x1.a = t1.a and x1.a = x2.a) from t1")
 	result.Check(testkit.Rows("1", "2", "3", "4", "5"))
+
+	tk.MustExec("drop table if exists t1, t2")
+	tk.MustExec("create table t1(a int)")
+	tk.MustExec("create table t2(b int)")
+	tk.MustExec("insert into t1 values(1)")
+	tk.MustExec("insert into t2 values(1)")
+	tk.MustQuery("select * from t1 where a in (select a from t2)").Check(testkit.Rows("1"))
```
Comment (XuHuaiyu) [inline executor/join_test.go]: Why t1.a = t2.a will be attached to EqualConds

Comment (lamxTyler) [reply executor/join_test.go]: Seems that the test does not cover the change?

Comment (winoros) [reply executor/join_test.go]: This test will cause master panicked

**Comment by XuHuaiyu** [comment]
LGTM

File: executor/join_test.go (hunk 1 above)
Comment (lamxTyler) [reply executor/join_test.go]: But I run successfully in master?

**Pull request merged**